state_machine/applications/flight/lib/mathutils.py

```diff
@@ -0,0 +1,40 @@
+from numpy import identity
+
+
+try:
+    from ulab.numpy import array, block, ndarray
+except Exception:
+    from numpy import array, block, ndarray
+
+def hat(v):
+    """Converts v to a matrix such that hat(v)w = cross(w, v) = -cross(v, w)"""
+    if not isinstance(v, ndarray):
+        v = array(v)
+    if v.shape == (3, 1):
+        return array([
+            [0,         -v[2][0],   v[1][0]],
+            [v[2][0],    0,        -v[0][0]],
+            [-v[1][0],   v[0][0],   0]])
+    elif v.shape == (3,):
+        return array([
+            [0, -v[2], v[1]],
+            [v[2], 0, -v[0]],
+            [-v[1], v[0], 0]])
+    else:
+        raise ValueError("v must be a 3x1 numpy array")
+
+def L(q):
```

Comment:
This is more of a style thing but I would prefer to have a more descriptive name for this function (ex. ScalarFirst_toLeft or scalar_first_to_left)

Reply:
Good idea

Reply:
And I'm actually ok with keeping it `L(q)` as well, assuming that we have good documentation of what the function does and means. This is a bit of a gray area since it has mathematical meaning to write it this way.

Reply:
I changed it to `quaternion_to_left_matrix`